Assets/Mirror/Tests/WeaverAssembler.cs

```diff
@@ -0,0 +1,188 @@
+    {
+        foreach (string src in sourceFiles)
+        {
+            SourceFiles.Add(OutputDirectory + src);
+        }
+    }
+
+    // Add a range of reference files by full path
+    public static void AddReferencesByFullPath(string[] refAsms)
+    {
+        foreach (string asm in refAsms)
+        {
+            ReferenceAssemblies.Add(asm);
+        }
+    }
+
+    // Add a range of reference files by assembly name only
+    public static void AddReferencesByAssemblyName(string[] refAsms)
+    {
+        foreach (string asm in refAsms)
+        {
+            string asmFullPath;
+            if (FindReferenceAssemblyPath(asm, out asmFullPath))
+            {
+                ReferenceAssemblies.Add(asmFullPath);
+            }
+        }
+    }
+
+    // Find reference assembly specified by asmName and store its full path in asmFullPath
+    // do not pass in paths in asmName, just assembly names
+    public static bool FindReferenceAssemblyPath(string asmName, out string asmFullPath)
+    {
+        asmFullPath = "";
+
+        Assembly[] asms = CompilationPipeline.GetAssemblies();
+        foreach (Assembly asm in asms)
+        {
+            foreach (string asmRef in asm.compiledAssemblyReferences)
+            {
+                if (asmRef.EndsWith(asmName))
+                {
+                    asmFullPath = asmRef;
+                    return true;
+                }
+            }
+        }
+
+        return false;
+    }
+
+    // Add reference (not cleared during calls to Clear)
+    public static void ClearReferences()
+    {
+        ReferenceAssemblies.Clear();
+    }
+
+    // Delete output dll / pdb / mdb
+    public static void DeleteOutput()
+    {
+        if (OutputFile.Length < 5) return; // "x.dll" shortest possible dll name
+
+        string projPathFile = OutputDirectory + OutputFile;
+
+        try
+        {
+            File.Delete(projPathFile);
+
+        }
+        catch { }
+
+        try
+        {
+            File.Delete(Path.ChangeExtension(projPathFile, ".pdb"));
+
+        }
+        catch { }
+
+        try
+        {
+            File.Delete(Path.ChangeExtension(projPathFile, ".mdb"));
+
+        }
+        catch { }
+    }
+
+    // clear all settings except for referenced assemblies (which are cleared with ClearReferences)
+    public static void Clear()
+    {
+        if (DeleteOutputOnClear)
+        {
+            DeleteOutput();
+        }
+
+        CompilerErrors = false;
+        OutputFile = "";
+        SourceFiles.Clear();
+        CompilerMessages.Clear();
+        AllowUnsafe = false;
+        DeleteOutputOnClear = false;
+    }
+
+    // build synchronously
+    public static void Build()
+    {
+        BuildAssembly(true);
+    }
+
+    // build asynchronously - this isn't currently used
+    public static void BuildAsync()
+    {
+        BuildAssembly(false);
+    }
+
+    private static void BuildAssembly(bool wait)
```

Comment:
make this async instead of passing wait

Reply:
https://github.com/vis2k/Mirror/pull/490/commits/b5d614bd95c17cbc39f76c5519e912d221e81514
Seems it broke everything. Am I doing something wrong?

Reply:
Reverted async commit, and everything is now working. Do you really want this async change on this PR?

Reply:
In order to be able to merge it, will try to make this async instead of bool wait (if unity is okay with that, huh) in future PR